### ISSUE TITLE
malloc network, but when run it, some wheres break!!!

### ISSUE THREAD

**Issue by xinsuinizhuan** [josephjaspers/blackcat_tensors#37]
I declare this:
auto make_lstm_network() {
	return BC::nn::neuralnetwork(
		BC::nn::lstm(BC::host_tag(), 96 * 10, 128),
		BC::nn::recurrent(BC::host_tag(), 128, 96),
		BC::nn::softmax(BC::host_tag(), 96),
		BC::nn::output_layer(BC::host_tag(), 96)
	);
}
using network_type = decltype(make_lstm_network());

typedef struct _LstmPredictTask {
	int input;
	int output;
	network_type m_pnetwork = make_lstm_network();
	void reset_neural_network() {
		m_pnetwork = std::move(make_lstm_network());
	}
} LstmPredictTask;

and when run this code, break at the funciton get_string_architecture() and  set_batch_size(32):
        LstmPredictTask* lstmpredicttask = (LstmPredictTask*)malloc(sizeof(LstmPredictTask));
	if (lstmpredicttask == NULL) {
		return -2;
	}

	//LstmPredictTask lstmpredicttask;
	auto network = lstmpredicttask->m_pnetwork;
	std::cout << "Neural Network architecture: \n" <<network.get_string_architecture() << std::endl;

	network.set_learning_rate(0.001);
	network.set_batch_size(batch_size);

I only malloc the network, then it break! 
when i do not set the set_batch_size, then for train, the function forward_propagation break.

**Comment by josephjaspers** [comment]
I must have missed this, I will take a look soon

**Comment by xinsuinizhuan** [comment]
> 
> 
> I must have missed this, I will take a look soon

when malloc network, then network.get_string_architecture(), the print network architecture is null and empty, and when train break, you can try malloc network, then get_string_architecture() , set_batch_size(batch_size) and train it

**Comment by josephjaspers** [comment]
-->
When you use malloc to create a network you simply allocate memory for the space of a neural network object, however malloc does not call the constructor which means the memory is left an uninitialized state. 

You should prefer to use `new` and call the constructor.

**Comment by xinsuinizhuan** [comment]
> new

OK, when i use 'new' it is work. Thank you!

**Comment by josephjaspers** [comment]
Awesome! Glad it worked.
